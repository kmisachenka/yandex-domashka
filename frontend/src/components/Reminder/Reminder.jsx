import React from 'react';
import ReactSVG from 'react-svg';
import * as PropTypes from 'prop-types';
import * as luxon from 'luxon';

import ReminderSVG from './svg/reminder.svg';
import styles from './Reminder.module.scss';

export default function Reminder(props) {
  const { reminder, className } = props;
  const time = luxon.DateTime.fromMillis(reminder).toRelative({ locale: 'ru' });
  /* dirty hack: вырезал 'через' */
  const hackedTime = time.substring(6, time.length);
  return (
    <section className={className}>
      <ReactSVG src={ReminderSVG} />
      <div className={styles.text}>{`Осталось ${hackedTime}`}</div>
    </section>
  );
}

Reminder.defaultProps = {
  className: '',
};

Reminder.propTypes = {
  reminder: PropTypes.number.isRequired,
  className: PropTypes.string,
};
