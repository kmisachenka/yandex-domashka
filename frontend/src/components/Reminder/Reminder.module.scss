.text {
  font-family: Helvetica, serif;
  font-size: 16px;
  line-height: 22px;
  font-weight: bold;

  color: #fff;

  padding-left: 14px;
}
