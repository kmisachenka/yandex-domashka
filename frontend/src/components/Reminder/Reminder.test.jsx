import React from 'react';
import { shallow } from 'enzyme';
import * as luxon from 'luxon';

import Reminder from './Reminder';

jest.mock('luxon');

luxon.DateTime.fromMillis = jest.fn(() => ({
  toRelative: jest.fn(() => 'через 2 дня'),
}));

describe('Reminder', () => {
  it('should match the snapshot', () => {
    const wrapper = shallow(<Reminder reminder={12345} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should apply custom classes', () => {
    const wrapper = shallow(<Reminder reminder={12345} className="custom" />);
    expect(wrapper).toHaveClassName('custom');
  });
});
