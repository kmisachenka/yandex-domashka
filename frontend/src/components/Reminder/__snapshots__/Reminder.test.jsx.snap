// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reminder should match the snapshot 1`] = `
<section
  className=""
>
  <ReactSVG
    evalScripts="never"
    fallback={null}
    loading={null}
    onInjected={[Function]}
    renumerateIRIElements={true}
    src="reminder.svg"
    svgClassName={null}
    svgStyle={Object {}}
    wrapper="div"
  />
  <div
    className="text"
  >
    Осталось 2 дня
  </div>
</section>
`;
